### ISSUE TITLE
About conf_loss_set_indices & exclude_loss_set_indices

### ISSUE THREAD

**Issue by jianglh-WHU** [facebookresearch/map-anything#72]
Hi! MapAnything is an excellent piece of work. I’m curious about the training loss design.

In your implementation, you set `conf_loss_set_indices=[0] and exclude_loss_set_indices=[1, 2]` to ignore imperfections and potential outliers. As I understand it, this means only the first three losses (some pts loss or cam_pts loss) are affected. Why not apply conf_loss and exclude_loss to all pts and cam_pts losses? Do you have any insights on this choice? If I apply conf_loss and exclude_loss to all pts and cam_pts losses, would performance be affected?

By the way, I also have a question about the nonAmbiguousMask. Is it mainly intended for sky prediction? If I don’t use this loss, or even don’t predict the nonAmbiguousMask, will the performance be affected?

Thanks!

**Comment by jianglh-WHU** [comment]
Oh, I understand that, you apply conf_loss to pts3d loss and apply exclude_loss to cam_pts loss & depth_along_ray loss. I have another question, could you tell me why you use the sum of losses rather than the mean of losses (view dimension)? If I use the mean of losses and keep the loss weight unchanged, would is the performance will be affected?

**Comment by jianglh-WHU** [comment]
Raydirection loss is also confused. Since the ray direction shouldn't be related to the valid mask, why is the mask being applied to this loss?

**Comment by qsisi** [comment]
> Raydirection loss is also confused. Since the ray direction shouldn't be related to the valid mask, why is the mask being applied to this loss?

Looks like the mask is not applied to the ray direction loss?
https://github.com/facebookresearch/map-anything/blob/44cbc5a4a3960cab50c449797afef200b88b94b7/mapanything/train/losses.py#L3079-L3083

**Comment by jianglh-WHU** [comment]
You are right, the mask is not applied to the ray direction loss. Thanks!

**Comment by Nik-V9** [comment]
Hi, thanks for your interest! Answers to your questions:

1. `conf_loss_set_indices=[0] and exclude_loss_set_indices=[1, 2]` are applied for every view. Here the 0, 1, 2 indicate the loss factor i.e. pts3d, depth, cam_pts3d, etc.
2. nonAmbiguousMask is primarily meant for sky prediction and not training it will not result in any performance degradation.
3. Sum vs mean: This isn't a big deal and mainly changes what the learning rate should be set to.
4. Yes, you are right, validity mask is not applied to ray directions since the ray value for every pixel is valid and is derived from the camera intrinsics.
